### PR TITLE
Pear module

### PR DESCRIPTION
**Pull request by leafnode**
New packaging module for PEAR repositories. Includes installing/uninstalling packages, as well as managing package channels.

### CODE REVIEW

**Comment by leafnode** [comment]
Any comments...?

File: packaging/language/pear.py [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+
+def _get_full_name(name, version=None):
+    if version is None:
+        resp = name
+    else:
+        resp = name + '-' + version
+    return resp
+
+def _is_package_present(module, name, version):
+    rc, out_pear, err_pear = module.run_command('pear list')
```
Comment (bcoca) [inline packaging/language/pear.py:82]: don't rely on path, use module's find bin to locate the correct pear executable (same in other instances below)

Comment (bcoca) [inline packaging/language/pear.py:82]: when running command you should also catch exceptions

File: packaging/language/pear.py [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+import re
+
+DOCUMENTATION = '''
+---
+module: pear
+short_description: Manages PHP PEAR library dependencies.
+description:
+     - "Manage PHP PEAR dependencies. To use this module, one of the following keys is required: C(name)
+       or C(channel)."
+version_added: "0.1"
```
Comment (bcoca) [inline packaging/language/pear.py:31]: 2.0

File: packaging/language/pear.py [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+options:
+  name:
```
Comment (bcoca) [inline packaging/language/pear.py:33]: ~~why is this not required? does the module do anything w/o name?~~
nvmd, just saw channel, you might still want to require on or the other (actual module function) and hint to that in docs

**Comment by leafnode** [comment]
Awesome, thanks for the notes, I'm getting to fixing that.

**Comment by gregdek** [comment]
@leafnode apologies, but there were multiple Pear modules submitted, and because another module passed review first, that module was accepted. Closing this PR.

**Pull request closed without merging**